src/app/constructor/constructor.component.ts

```diff
@@ -26,13 +43,42 @@ export class ConstructorComponent {
         this.canBeMapActive = !value;
     }
 
-    public onDashboardDefaultChange(value: string) {
-        this.default = value;
+    public onSaveView(): void {
+        if (!this.isViewCanBeSaved()) {
+            this.toastr.error('Please fill mandatory fields: "Name", "Description" and (("Add Picture") or ("Dashboard"))');
+            return;
+        }
+        this.viewService.create(this.view).subscribe(({mapSubview}) => {
+            if (this.uploader) {
+                return this.uploadPicture(mapSubview);
+            }
+            this.router.navigate(['..']);
+        });
     }
 
-    public onMapDefaultChange(value: string) {
-        this.default = value;
+    public onUploadPicture(uploader: FileUploader) {
+        this.uploader = uploader;
     }
 
-    public onSaveView(): void {}
+    private isViewCanBeSaved(): boolean {
+        return this.view.name
```

Comment:
Just suggestion to simplify isViewCanBeSaved extacting some logic to separate method:
return this.view.name
&& this.view.description
&& this.pictureUploadedOrdeviceAdded()